lib/ansible/plugins/inventory/foreman.py

```diff
@@ -231,6 +246,30 @@ def _populate(self):
                 if self.get_option('want_facts'):
                     self.inventory.set_variable(host_name, 'foreman_facts', self._get_facts(host))
 
+                # create group for host collections
+                if self.get_option('want_hostcollections'):
+                    host_data = self._get_host_data_by_id(host['id'])
```

Comment:
At some point, this will probably be locally cached, as it's done in the script

https://github.com/ansible/ansible/blob/4311990c3312b23e87d26c86ebca394501aecd4a/lib/ansible/plugins/inventory/foreman.py#L251

but that intersects the development of reports, https://github.com/ansible/ansible/pull/62438, so it makes sense that's not done right now.